### PR TITLE
Dev/1d reduction

### PR DESCRIPTION
**Pull request by avolkov-intel**
Add 1d reducer implementation to reduce primitives

### CODE REVIEW

**Review by KulikovNikita — APPROVED** [review]

**Comment by KulikovNikita** [comment]
Please start Private CI on this branch

File: cpp/oneapi/dal/backend/primitives/reduction/test/reduction_1d_dpc.cpp [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+
+#include "oneapi/dal/backend/primitives/reduction/reduction_1d.hpp"
+
+namespace oneapi::dal::backend::primitives::test {
+
+namespace te = dal::test::engine;
+namespace pr = oneapi::dal::backend::primitives;
+
+using std::cout;
+using std::endl;
```
Comment (KulikovNikita) [inline cpp/oneapi/dal/backend/primitives/reduction/test/reduction_1d_dpc.cpp:35]: Please delete

**Review by Alexsandruss — APPROVED** [review]

File: cpp/oneapi/dal/backend/primitives/reduction/test/reduction_1d_dpc.cpp [hunk 2]
```diff
@@ -0,0 +1,106 @@
+/*******************************************************************************
+* Copyright 2022 Intel Corporation
```
Comment (KulikovNikita) [inline cpp/oneapi/dal/backend/primitives/reduction/test/reduction_1d_dpc.cpp:2]:
```suggestion
* Copyright 2023 Intel Corporation
```

File: cpp/oneapi/dal/backend/primitives/reduction/reduction_1d_dpc.cpp [hunk 3]
```diff
@@ -0,0 +1,97 @@
+/*******************************************************************************
+* Copyright 2022 Intel Corporation
```
Comment (KulikovNikita) [inline cpp/oneapi/dal/backend/primitives/reduction/reduction_1d_dpc.cpp:2]:
```suggestion
* Copyright 2023 Intel Corporation
```

File: cpp/oneapi/dal/backend/primitives/reduction/reduction_1d.hpp [hunk 4]
```diff
@@ -0,0 +1,101 @@
+/*******************************************************************************
+* Copyright 2022 Intel Corporation
```
Comment (KulikovNikita) [inline cpp/oneapi/dal/backend/primitives/reduction/reduction_1d.hpp:2]:
```suggestion
* Copyright 2023 Intel Corporation
```

**Review by KulikovNikita — APPROVED** [review]

**Pull request merged**